development/external-plugins/test-untrusted/cmd/main.go

```diff
@@ -39,19 +38,34 @@ func EventHandler(server *externalplugin.Plugin, event externalplugin.Event) {
 		"pr-sender", pr.Sender.Login,
 	)
 	switch pr.Action {
-	case github.PullRequestActionOpened:
+	case github.PullRequestActionOpened, github.PullRequestActionReopened, github.PullRequestActionSynchronize, github.PullRequestActionEdited:
```

Comment:
I don't think you need to listen to `github.PullRequestActionReopened`. If the PR is reopened without updates the statuses are retained, so there's no need to run the tests again. If there happen to be updates you will get `github.PullRequestActionSynchronize` as a separate event.

Reply:
I did test scenario when pr head branch got commit between closing and reopening it. As you wrote there was a `github.PullRequestActionSynchronize` event present on a deliveries list.

However I built list of pull_request actions following actions which are supported by trigger plugin. https://github.com/kubernetes/test-infra/blob/d1f247f5d7711e2f77af97fc5fdc91dce5928319/prow/plugins/trigger/pull-request.go#L93

The question is if there is any scenario when we need to react on this event.

Reply:
Ok, found it. If your PR is closed, then some commit will be merged in to base branch next you will reopen a PR. There is only `github.PullRequestActionReopened` event, no `github.PullRequestActionSynchronize` is generated.